brax/training/agents/ppo/train.py

```diff
@@ -231,12 +231,16 @@ def train(
   key_envs = jnp.reshape(key_envs,
                          (local_devices_to_use, -1) + key_envs.shape[1:])
   env_state = reset_fn(key_envs)
+  ndarray_obs = isinstance(env_state.obs, jnp.ndarray) # Check whether observations are in dictionary form.
+
+  obs_shape = env_state.obs.shape[-1] if ndarray_obs \
+    else jax.tree_util.tree_map(lambda x: x.shape[2:], env_state.obs) # Discard batch axes.
```

Comment:
Why do we have [2:] for the dict, buy [-1] for the jax.Array version? Presumably for pixel inputs?

Can they both be [2:] and then remove the if statement?

Can the comment be more explicit about why the first two dims are removed (one for num_envs, and one for num_devices, is that right?)